src/components/ThemeProvider.tsx

```diff
@@ -0,0 +1,54 @@
+import { useMemo } from "react"
+import merge from "lodash/merge"
+import { ThemeProvider as NextThemesProvider } from "next-themes"
+import type { ThemeProviderProps } from "next-themes/dist/types"
+import { ChakraBaseProvider, createLocalStorageManager } from "@chakra-ui/react"
+
+import customTheme from "@/@chakra-ui/theme"
+
+import { COLOR_MODE_STORAGE_KEY } from "@/lib/constants"
+
+import { useLocaleDirection } from "@/hooks/useLocaleDirection"
+import { mono } from "@/lib/fonts"
+
+const colorModeManager = createLocalStorageManager(COLOR_MODE_STORAGE_KEY)
+
+/**
+ * Primary theming wrapper for use with color mode. Uses the theme provider
+ * from `next-themes`.
+ *
+ * Applied to _app.tsx as the main provider for the project, and supplied as the
+ * primary decorator to Storybook.
+ *
+ * NOTE: This also includes the Chakra Provider. This will be removed after migration to ShadCN/Tailwind is complete
+ */
+const ThemeProvider = ({ children }: Pick<ThemeProviderProps, "children">) => {
+  const direction = useLocaleDirection()
+
+  const theme = useMemo(() => merge(customTheme, { direction }), [direction])
+  return (
+    <NextThemesProvider
```

Comment:
Having both providers in one reusable component helps with the sync with Storybook and prod.

Also note here that I added the style tag in here with the CSS vars for the fonts for the same reason as above. See that `TODO` comment also

Reply:
Looks better 👍🏼